jsonrpc/eth_endpoint_test.go

```diff
@@ -152,22 +152,28 @@ func TestEth_Block_GetBlockByNumber(t *testing.T) {
 
 	cases := []struct {
 		blockNum BlockNumber
+		isNotNil bool
 		err      bool
 	}{
-		{LatestBlockNumber, false},
-		{EarliestBlockNumber, true},
-		{BlockNumber(-50), true},
-		{BlockNumber(0), false},
-		{BlockNumber(2), false},
-		{BlockNumber(50), true},
+		{LatestBlockNumber, true, false},
+		{EarliestBlockNumber, false, true},
+		{BlockNumber(0), true, false},
+		{BlockNumber(2), true, false},
+		{BlockNumber(50), false, false},
```

Comment:
I noticed that you removed `-50` test. Do you think it's bad if we have it or was that an accident?

Reply:
I'm thinking whether we should return error, or not in this case. But actually negative number won't be passed to this function. (because parsing arguments is failed) It just returns null if negative number is passed in current implementation.

Reply:
I let function not to return error when transaction doesn't exist. For that reason, this function doesn't return error if -50 is passed. So I'm thinking if I should add new check to detect arg is negative.

Reply:
I tested with mainnet and they return error if first bit is set in hex arg. So I let this function return error in this case.

Reply:
@Vuksan If you agree with my opinion and the change, please just click `Resolve conversation`. Thanks